environment.yml

```
@@ -0,0 +1,10 @@
+name: dem
+channels:
+  - conda-forge
+  - nodefaults  # force everything to come from conda-forge channel
+dependencies:
+  - conda >= 24.0
+  - gdal >= 3.4.2
+  - python >= 3.11
```

Which dependency is forcing `python` to be >= 3.11? `sardem` has a [3.6 minimum](https://github.com/scottstanie/sardem/blob/master/environment.yml), and `scalene` seems to [support from 3.8 (but not 3.11.0)](https://github.com/plasma-umass/scalene/blob/8e78842508fdb47e268b15b31341adf71b38bb2c/setup.py#L233).

---

Nothing is forcing this, other than me. There are speed improvements to be gained with 3.11 and even 3.12, which may or may not yield significant differences in our stats. There are other benefits as well, although not necessary exercised with this small bit of code, such as improved stack traces and many other things that we may or may not benefit from with this algo.

Again, I generally like to use the latest stable version of Python when there's nothing in the way of doing so. And again, I don't have particularly strong feelings about this here either, so if you want me to pin to an earlier version, I'll do so. (By the way, scalene has no problems with 3.11 as evidenced by successful job runs.)

---

> There are speed improvements to be gained with 3.11 and even 3.12

Ah, I didn't realize that. If so, then we should probably considering fixing the version for benchmarking purposes, and noting the reasoning. Maybe something like:


```suggestion
  - python = 3.12  # algorithm only requires python >= 3.8, but fix to 3.12.?? for benchmarking
```

---

Makes sense to pin this, so I'll try 3.12, but if that causes a problem (for dependencies), I'll pin to 3.11.

---

Per my other comment regarding the need to change ESA's base image for the Dockerfile, I've pinned python to 3.10.12.